File: selection sort/selectionsort.py
def insertionSort(array: list) -> list:
    n = len(array)

    for i in range(n - 1):
        minIndex = i
        for j in range(i + 1, n):
            if array[j] < array[minIndex]:
                minIndex = j
            array[i], array[minIndex] = array[minIndex], array[i]
    return array
                
print(insertionSort([3,8,2,5,9,4,10]))

File: selection sort/selectionsort.js
function selectionSort(array){
    n = array.length

    for (let i = 0; i < n - 1; i++){
        minIndex = i
        for(let j = i + 1; j < n; j++){
            if(array[j] < array[minIndex]){
                minIndex = j
            }
            [array[minIndex], array[i]] = [array[i], array[minIndex]]
        }
    }
    console.log(array)
}

selectionSort([3,4,2,7,6,1,19,45])